### DISCUSSION TITLE
How to trouble-shooting high packets rate causing packet loss problem using IPSec VPN?

### DISCUSSION THREAD

**Discussion by jerryishe** [strongswan/strongswan#1069] [Q&A]
Hi Team,

We meet a problem when using IPSec VPN via Strongswan.

We create IPSec tunnel between Linux Centos VM and FortiGate firewall and it runs well. But we sometimes meet high packet rate and the IPSec Tunnel will get issue.

We monitor the outbound packet ratio to the remote firewall side and it hits to **200Kpps** by some CDN sevice. And while the packets ratio reaches some high level, it will suddenly drop back to a lowest value (20K) and traffic lose some packets. Meanwhile, clients will find the VPN tunnel drops any following traffic (Web service can not be access). 

I check the /var/log/message log and found nothing, someone tell me it's the preformance limitation and need to upgrade my VM to higher than 4 core 8GB. But I dont think it is to be blame on performance cuz the avg CPU usage is 30% and memory  usage is lower than 5%.

Please help me to **diagnoze what cause this packets loss.** And **how to fix that** (As I know, the service with high packet ratio can not be mamually banned)

Thank you 
jerry.

**Comment by tobiasbrunner** [comment]
Really difficult to say, in particular since there is a proprietary implementation involved (i.e. you might want to contact FortiGate if the issue is mostly sending packets to that box).

In general, one issue with large bursts of packets is the anti-replay protection. On Linux, the default is a replay window of only 32 packets (inbound packets with sequence numbers outside that window get dropped). So you definitely want to increase that via `charon.replay_window` in strongswan.conf or `<child>.replay_window` in swanctl.conf (there is a similar setting in ipsec.conf). It's a local setting that's not negotiated, so no idea if you can change it on the FortiGate side.

Your description could also indicate a queuing issue (i.e. new packets are dropped because the queue is full of packets that need to get processed first). But since that might be mostly on the FortiGate side, it's difficult to say if that's actually what's happening (again, you might want to contact them).

**Reply by jerryishe** [reply]
Hello, I have two questions on your reply.
1. I have checked that fortigate set default replay window as 2048, and I can adjust swan side to the same. But I am confused that **How can I know this parameter work on the VPN? And can I get any logs about replay issue?**
2. How can I see the packets queuing performance on Linux? And any solution to make an optimization? 

Thanks.

**Reply by tobiasbrunner** [reply]
From your description, I don't think Linux is the issue.
